### DISCUSSION TITLE
[잡담] 2주차 과제 3번 문제에 튜플을 사용했었네요!

### DISCUSSION THREAD

**Discussion by ghost** [4Moyede/HexatonClass-01#98] [잡담]
>###  코드
<pre>
<code>
def order_print(new_list_result, count):

new_list_result.sort(key = lambda x : (x[1], x[0])) 
    print("[ Sentence %d ]" % count)
    for word in new_list_result:
        print('{} : {}'.format(word[0], word[1]))
</code>
</pre>

<br>

>### 설명
제가 지난 2주차 과제 3번 문제에서 정의했던 함수들 중 하나입니다. <br>

_(key = lambda (x : x[1], x[0]))_ 라는 구절을 사용했는데요, 간단히 설명드리면 new_list_result 는 2차원 배열입니다.
1X2 크기의 행렬이 일렬 (위아래로) 쭉 늘어져 있는 형태입니다.<br>

배열의 첫번째 요소에는 str 형식이 들어갔고, 두번째 요소에는 int 형식이 들어갔습니다.
저는 두번째 요소인 int 형식의 오름차순으로 배열을 정리하고자 했는데, 여기서 튜플 형식을 사용했더라고요!
_(x[1], x[0])_ 바로 이겁니다!<br>
<br>
>### 반전
그런데 갑자기 그런 생각이 들었습니다.
그냥 _[x[1], x[0]]_ 처럼 배열 형식은 안되는걸까?
그래서 해봤습니다. 그러니까 되더라고요. 튜플 형식이 꼭 필요한 상황인가 생각했는데 아니었습니다.<br>

코드를 다시 읊어보니까 새로운 깨달음을 하나씩 알게 되는데 흥미롭네요.

**Comment by yoomambo** [comment]
@JaebeomHan 구우우우웃 👍
